Frends.Community.Pgp/Frends.Community.Pgp/PgpTasks.cs

```diff
@@ -0,0 +1,549 @@
+        /// </summary>
+        public static PgpClearTextSignatureResult PgpClearTextSignFile(PgpClearTextSignatureInput input)
+        {
+            HashAlgorithmTag digest;
+            if (input.HashFunction == PgpClearTextSignatureHashFunctionType.MD5)
+            {
+                digest = HashAlgorithmTag.MD5;
+            }
+            else if (input.HashFunction == PgpClearTextSignatureHashFunctionType.RipeMD160)
+            {
+                digest = HashAlgorithmTag.RipeMD160;
+            }
+            else if (input.HashFunction == PgpClearTextSignatureHashFunctionType.Sha1)
+            {
+                digest = HashAlgorithmTag.Sha1;
+            }
+            else if (input.HashFunction == PgpClearTextSignatureHashFunctionType.Sha224)
+            {
+                digest = HashAlgorithmTag.Sha224;
+            }
+            else if (input.HashFunction == PgpClearTextSignatureHashFunctionType.Sha384)
+            {
+                digest = HashAlgorithmTag.Sha384;
+            }
+            else if (input.HashFunction == PgpClearTextSignatureHashFunctionType.Sha512)
+            {
+                digest = HashAlgorithmTag.Sha512;
+            }
+            else
+            {
+                digest = HashAlgorithmTag.Sha256;
+            }
+
+            Stream privateKeyStream = File.OpenRead(input.PrivateKeyFile);
+
+            PgpSecretKey pgpSecKey = Services.ReadSecretKey(privateKeyStream);
+            PgpPrivateKey pgpPrivKey = pgpSecKey.ExtractPrivateKey(input.Password.ToCharArray());
+            PgpSignatureGenerator sGen = new PgpSignatureGenerator(pgpSecKey.PublicKey.Algorithm, digest);
+            PgpSignatureSubpacketGenerator spGen = new PgpSignatureSubpacketGenerator();
+
+            sGen.InitSign(Org.BouncyCastle.Bcpg.OpenPgp.PgpSignature.CanonicalTextDocument, pgpPrivKey);
+
+            IEnumerator enumerator = pgpSecKey.PublicKey.GetUserIds().GetEnumerator();
+            if (enumerator.MoveNext())
+            {
+                spGen.SetSignerUserId(false, (string)enumerator.Current);
+                sGen.SetHashedSubpackets(spGen.Generate());
+            }
+
+            Stream fIn = File.OpenRead(input.InputFile);
+            Stream outputStream = File.Create(input.OutputFile);
+
+            ArmoredOutputStream aOut = new ArmoredOutputStream(outputStream);
+
+            aOut.BeginClearText(digest);
+
+            //
+            // note the last \n/\r/\r\n in the file is ignored
+            //
+            MemoryStream lineOut = new MemoryStream();
+            int lookAhead = Services.ReadInputLine(lineOut, fIn);
+
+            Services.ProcessLine(aOut, sGen, lineOut.ToArray());
+
+            if (lookAhead != -1)
+            {
+                do
+                {
+                    lookAhead = Services.ReadInputLine(lineOut, lookAhead, fIn);
+
+                    sGen.Update((byte)'\r');
```

Comment:
Later the Environment.NewLine is being used. Would it be correct here as well? Or should it be configurable by the user?